build_tools/i18n/utils.py

```diff
@@ -52,6 +51,19 @@
 }
 
 
+def to_locale(language):
+    """
+    Turns a language name (en-us) into a locale name (en_US).
+    """
+    p = language.find('-')
+    if p >= 0:
+        if len(language[p + 1:]) > 2:
+            return language[:p].lower() + '_' + language[p + 1].upper() + language[p + 2:].lower()
```

Comment:
what is an example where this code would get run?

could we simplify this function to something like 

```python
codes = language.split("-");
output = codes[0];
if len(codes) == 2:
    return "{}_{}".format(codes[0], codes[1].upper())
return codes[0]
```

or does that not handle all cases?

Reply:
This is directly copied from the Django source, the only thing I removed was the `lower` optional kwarg, which renders the second half into lower case.

The second part of the lang code can be title case, if the second piece is a script code, e.g. zh_Hans or zh_Hant, hence the conditional behaviour if the second part is longer than 2 characters.